cookies/views.py

```diff
@@ -142,11 +142,18 @@ def collection(request, obj_id):
     resources = collection.resources.filter(content_resource=False, hidden=False)
     resources = authorization.apply_filter(request.user, 'view_resource', resources)
     filtered_objects = ResourceFilter(request.GET, queryset=resources)
-
+    qset_collections = Collection.objects.filter(
```

Comment:
It looks like what you're doing here is getting all of the Collections in the database, filtering like we do for the list view (so, still most collections), applying a filter (not sure what that's for?), and then in the template iterating over all of the collections and only displaying them if they are part of the current collection. Is that right?

So, that's a looooot of database overhead. What if we have 10,000 collections, only two of which are sub-collections of the current collection?

Take a look at [these docs](https://docs.djangoproject.com/en/1.10/ref/models/relations/) for clues to a better way. :-)